src/query/api/v1/handler/placement/init.go

```diff
@@ -101,7 +110,18 @@ func (h *InitHandler) Init(
 		return nil, err
 	}
 
-	service, err := Service(h.client, httpReq.Header)
+	serviceOpts := NewServiceOptionsFromHeaders(serviceName, httpReq.Header)
+	if serviceName == M3AggServiceName {
+		if req.MaxAggregationWindowSizeNanos == 0 || req.WarmupDurationNanos == 0 {
+			return nil, errAggWindowAndWarmupMustBeSet
```

Comment:
Hm seems this code is duplicated elsewhere?  Can we reuse?  Also same question about whether we can have defaults for these.